lib/proxy_domain.js

```diff
@@ -0,0 +1,359 @@
+const session = require("cookie-session");
+const cookieParser = require("cookie-parser");
+const HttpProxy = require("./proxy");
+const log = require("./log");
+const http = require("http");
+const Router = require("express").Router;
+const passport = require("passport");
+const rp = require("request-promise-native");
+const LocalStrategy = require("passport-local-token").Strategy;
+const GitHubStrategy = require("passport-github").Strategy;
+const GoogleStrategy = require("passport-google-oauth").OAuth2Strategy;
+
+const verifyEmailDomain = (valids, profile) => {
+  let emailDomains = profile.emails.map(e => e.value.split("@")[1]);
+  return valids.some(domain => emailDomains.some(emailDomain => domain == emailDomain));
+};
+
+const verifyEmail = (valids, profile) => {
+  let emails = profile.emails.map(e => e.value);
+  return valids.some(validEmail => emails.some(email => email == validEmail));
+};
+
+const verifyGitHubOrganization = (valids, profile) => {
+  let orgs = profile.organizations;
+  return valids.some(validOrg => orgs.some(org => validOrg == org));
```

Comment:
I think we may not need `some` on the inner one:
```suggestion
  return orgs.some(org => valids.includes(org));
```

Reply:
And done